clients/cobol-lsp-vscode-extension/src/extension.ts

```diff
@@ -92,11 +93,14 @@ export async function activate(context: vscode.ExtensionContext) {
             new CopybooksCodeActionProvider(profileService)));
 }
 
-function initWorkspaceTracker(downloader: CopybooksDownloader): vscode.Disposable {
+function initWorkspaceWatcher(downloader: CopybooksDownloader): vscode.Disposable {
     const watcher = vscode.workspace.createFileSystemWatcher("**/"
-        + DEPENDENCIES_FOLDER + "/**/**.dep", false, false, true);
+        + path.join(C4Z_FOLDER, DEPENDENCIES_FOLDER) + "/**/**.dep", false, false, true);
     watcher.onDidCreate(uri => downloader.downloadDependencies(uri,
         "Program contains dependencies to missing copybooks."));
     watcher.onDidChange(uri => downloader.downloadDependencies(uri));
+
+    //create .gitignore file within .c4z folder
+    createFileWithGivenPath(C4Z_FOLDER, GITIGNORE_FILE, "/**");
```

Comment:
Did you try the case if the user opens a new folder or change branch?

Reply:
Yes, for changing ws ok, and branch change has no impact, app works fine with old hierarchy